src/Actions/EmailWorkflowAction.php

```diff
@@ -0,0 +1,138 @@
+        'EmailSubject' => 'Varchar(256)',
+    ];
+
+    private static $has_one = [
+        'EmailTemplate' => UserTemplate::class,
+        'EmailMember' => Member::class,
+        'EmailGroup' => Group::class,
+    ];
+
+    private static $table_name = 'EmailWorkflowAction';
+
+    public function getCMSFields()
+    {
+        $fields = parent::getCMSFields();
+
+        $newFields = FieldGroup::create('Email config');
+        $newFields->push(LiteralField::create('CSS', '<style>.form__fieldgroup-item { min-width: 30% !important; }</style>'));
+
+        // type
+        $types = $this->dbObject('EmailTarget')->enumValues();
+        $newFields->push(DropdownField::create('EmailTarget', $this->fieldLabel('EmailTarget'), $types));
+        // from
+        $newFields->push(TextField::create('EmailFrom', $this->fieldLabel('EmailFrom')));
+        // subject
+        $newFields->push(TextField::create('EmailSubject', $this->fieldLabel('EmailSubject')));
+        // template
+        $templates = UserTemplate::get()->filter(['Use' => 'Layout']);
+        $newFields->push(DropdownField::create('EmailTemplateID', $this->fieldLabel('EmailTemplateID'), $templates)
+            ->setEmptyString('-- default --'));
+        // to
+        $toTitle = _t('EmailWorkflowAction.EMAILTO', 'Send to');
+        if ($this->EmailTarget === 'Manual') {
+            $newFields->push(EmailField::create('EmailManual', $toTitle)->setDescription('Normal email address'));
+        }
+        else if ($this->EmailTarget === 'Field') {
+            $newFields->push(TextField::create('EmailField', $toTitle)->setDescription('Name of target field'));
+        }
+        else if ($this->EmailTarget === 'Member') {
+            $newFields->push(DropdownField::create('EmailMemberID', $toTitle, Member::get()->map()));
+        }
+        else if ($this->EmailTarget === 'Group') {
+            $newFields->push(DropdownField::create('EmailGroupID', $toTitle, Group::get()->map()));
+        }
+
+        $fields->insertBefore('Property', $newFields);
+        $this->extend('updateTargetMethodCMSFields', $fields);
+        return $fields;
+    }
+
+    public function fieldLabels($relations = true)
+    {
+        return array_merge(parent::fieldLabels($relations), [
+            'EmailTarget' => _t('EmailWorkflowAction.EMAILTARGET', 'Target type'),
+            'EmailFrom' => _t('EmailWorkflowAction.EMAILFROM', 'Send from'),
+            'EmailSubject' => _t('EmailWorkflowAction.EMAILSUBJECT', 'Email subject'),
+            'EmailTemplateID' => _t('EmailWorkflowAction.EMAILTEMPLATE', 'Email template'),
+        ]);
+    }
+
+    public function execute(WorkflowInstance $workflow)
+    {
+        parent::execute($workflow);
+
+        $target = $workflow->getTarget();
+        $from = $this->EmailFrom;
+        $subject = $this->EmailSubject;
+        $template = $this->EmailTemplate();
+
+        if (!$target || !$from || !$subject || !$template) {
+            return false;
```

Comment:
I think the default case here should be to return "true" so that the workflow will progress even if something's gone wrong in the configuration of the workflow.